Automattic/sensei
# Add HPPS settings integration

Resolves #7066

- I didn't replace the feature flag with this setting (actually, that was my first take, but I reverted it), instead I try to manage the flag in the Sensei_Main class depending on the setting value.
- We don't take the UI setting into account when checking the HPPS setting, i.e. if the UI feature flag is not enabled we still apply HPPS setting value (enable/disable HPPS depending on the setting value).
- We create custom tables only at the moment we enable HPPS (when it transitions from disabled to enabled).


## Proposed Changes

* Handle the HPPS setting. 

## Testing Instructions

1. Drop custom progress tables.
2. Go to Sensei LMS Settings > Experimental Features.
3. Enable HPPS.
4. Make sure custom tables created.
5. Take a course, submit a quiz.
6. Check data appeared in custom tables.

## Pre-Merge Checklist
